r-package/grf/bindings/RegressionForestBindings.cpp

```diff
@@ -47,6 +48,9 @@ Rcpp::List regression_train(Rcpp::NumericMatrix train_matrix,
 
   Data* data = RcppUtilities::convert_data(train_matrix, sparse_train_matrix);
   data->set_outcome_index(outcome_index - 1);
+  if(sample_weight_index - 1 != data->get_num_cols()) {
```

Comment:
I find this approach to indicating whether sample weights are present to be a bit tricky/ fragile. Ideally we'd use `nonstd::optional<size_t>` class, but I don't think it'll work in an Rcpp exported method. Maybe we could use a `size_t *` here, and set it to `NULL` if sample weights aren't used (this is a common way of expressing 'optional' in C).

Reply:
I don't like it either, but it was the best I could come up with. The other idea I thought might be reasonable is to make the index signed and pass -1 as missing, but it seemed odd to have one signed index. The possibly-null pointer approach would be pretty awkward on the R side.

 or the C++ side but unnatural on the other. Passing an R list of parameters instead of multiple arguments is another.

Reply:
Got it, in that case could we introduce a boolean `contains_sample_weights`? It is quite clunky, but I think safety + clarity is more important than elegance.

Reply:
Good call.